core/src/main/java/org/frankframework/parameters/ParameterValue.java

```diff
@@ -157,6 +144,12 @@ private String valueAsString() {
 				throw new IllegalStateException("cannot open stream", e);
 			}
 		}
+
+		if (getDefinition() instanceof DateParameter dateParameter && value instanceof Date date) {
```

Comment:
Moeten we class van definiton checken, of ParameterType?

Reply:
De ParameterType checken en daarin de instance checken is waarschijnlijk sneller. Het vereist echter wel developer discipline, omdat de lijst met datum gerelateerde types kan veranderen. Als dit niet gelijk loopt kan het een subtiele bug introduceren. Wat heeft jouw voorkeur?

Reply:
In theorie zou iemand in code een instance van type `Parameter` kunnen maken en een datum-gerelateerd type daarop kunnen zetten, maar ik heb even gezocht in de code en dat gebeurt nergens.
Ik denk dat in praktijk de `instanceof` - check toch wel genoeg zal zijn.